crates/compiler/can/src/suffixed.rs

```diff
@@ -316,21 +249,15 @@ pub fn unwrap_suffixed_expression_apply_help<'a>(
             }
 
             // special case for when our Apply function is a suffixed Var (but not multiple suffixed)
-            if let Expr::Var { module_name, ident, suffixed } = function.value {
-                if suffixed == 1 {
-                    let unwrapped_function = arena.alloc(Loc::at(
-                        loc_expr.region,
-                        Expr::Var {
-                            module_name,
-                            ident,
-                            suffixed: suffixed - 1,
-                        },
-                    ));
+            if let Expr::TaskAwaitBang(sub_expr) = function.value {
+                let unwrapped_function = arena.alloc(Loc::at(
+                    loc_expr.region,
+                    *sub_expr,
+                ));
 
-                    let new_apply = arena.alloc(Loc::at(loc_expr.region, Expr::Apply(unwrapped_function, local_args, called_via)));
+                let new_apply = arena.alloc(Loc::at(loc_expr.region, Expr::Apply(unwrapped_function, local_args, called_via)));
 
-                    return init_unwrapped_err(arena, new_apply, maybe_def_pat);
-                }
+                return init_unwrapped_err(arena, new_apply, maybe_def_pat);
```

Comment:
We no longer need to match on `Expo::Var`, instead we can unwrap `Expr::TaskAwaitBang` directly. This is much cleaner and simplifies things.